libraries/chain/controller.cpp

```diff
@@ -4066,6 +4066,15 @@ struct controller_impl {
                   "QC is_strong (${s1}) in block extension does not match is_strong_qc (${s2}) in header extension. Block number: ${b}",
                   ("s1", qc_proof.is_strong())("s2", new_qc_claim.is_strong_qc)("b", block_num) );
 
+      if (prev.valid) {
+         // compute finality mroot using previous block state and new qc claim
+         auto computed_finality_mroot = prev.get_finality_mroot_claim(new_qc_claim);
+         auto supplied_action_mroot = b->action_mroot;
```

Comment:
No need to make a copy.

Reply:
It was used twice; that's why I created a copy. But it is used in the same EOS_ASSERT; will remove it for simplicity.

Reply:
You could use `const auto& supplied_action_mroot = b->action_mroot;`

Reply:
Thanks. That will avoid copying 32 bytes.